### PR TITLE
Support _Complex in the C++ front-end

### PR DESCRIPTION
**Pull request by tautschnig**
We typecheck this via the same code path as the C front-end, and just
need to accept that this type exists.

Clang-14 uses _Complex in system headers, and moving to Ubuntu 22.04
GitHub runners requires fixing this first.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/cpp/cpp_convert_type.cpp [hunk 1]
```diff
@@ -330,15 +330,9 @@ void cpp_convert_plain_type(typet &type, message_handlert &message_handler)
     type.id() == ID_union || type.id() == ID_array || type.id() == ID_code ||
     type.id() == ID_unsignedbv || type.id() == ID_signedbv ||
     type.id() == ID_bool || type.id() == ID_floatbv || type.id() == ID_empty ||
-    type.id() == ID_constructor || type.id() == ID_destructor)
+    type.id() == ID_constructor || type.id() == ID_destructor || type.id() == ID_c_enum)
   {
   }
-  else if(type.id()==ID_c_enum)
```
Comment (kroening) [inline src/cpp/cpp_convert_type.cpp:336]: This change seems unrelated to _Complex?

Comment (tautschnig) [reply src/cpp/cpp_convert_type.cpp:336]: Indeed, see the last sentence of the commit message in the second commit. It's drive-by cleanup while debugging to eventually make it possible to successfully pass the regression test that the second commit enables. (That regression test ensures we have coverage of the changes in the first commit.)

**Review by kroening — APPROVED** [review]
